src/tiled/variantpropertymanager.cpp

```diff
@@ -135,12 +141,59 @@ int VariantPropertyManager::alignmentTypeId()
     return qMetaTypeId<AlignmentPropertyType>();
 }
 
+QString VariantPropertyManager::objectRefLabel(const MapObject *object) const {
+    QString label = tr("%1: ").arg(QString::number(object->id()));
+    if (!object->name().isEmpty()) {
+        label.append(object->name());
+        if (!object->type().isEmpty())
+            label.append(tr(" (%1)").arg(object->type()));
+    } else if (!object->type().isEmpty())
+        label.append(tr("(%1)").arg(object->type()));
+    else
+        label.append(tr("Unnamed object"));
+
+    return label;
+}
+
 QString VariantPropertyManager::valueText(const QtProperty *property) const
 {
     if (mValues.contains(property)) {
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectRefTypeId()) {
+            ObjectRef ref = value.value<ObjectRef>();
+
+            if (ref.id == 0)
+                return tr("Unset");
+
+            Document *document = DocumentManager::instance()->currentDocument();
+            if (auto mapDocument = qobject_cast<MapDocument*>(document)) {
+                // Search all objects in the map.
+                for (const Layer *layer : mapDocument->map()->objectGroups()) {
+                    for (const MapObject *object : qobject_cast<const ObjectGroup*>(layer)->objects()) {
+                        if (object->id() == ref.id) {
```

Comment:
We should really not iterate all objects to find one by ID. We should be able to set up a `QHash<int, MapObject*>` in the `Map` class (not sure what to do about object references in object groups on tiles, though).

Reply:
Punted this to a new issue.